### DISCUSSION TITLE
Dropdown for known query parameters/variables names and values

### DISCUSSION THREAD

**Discussion by JMLX42** [scalar/scalar#2259] [Ideas]
Hi there!

Scalar looks crazy good! But there are some things that are keeping us from being able to use/support it.
So I'll try to document those as I find them.
Here is the first one.

## Feature request

Follow up on https://github.com/scalar/scalar/issues/525

The (required?) query parameters are initially listed:

![image](https://github.com/scalar/scalar/assets/998703/aaaf0e48-55e6-41eb-9017-ca437a97fe6a)

But (removing and then) adding a new query parameter requires to enter its name by hand. Since the query parameters are listed in the OpenAPI definition, I would expect them to be available in a dropdown.

As a reference, this is what Swagger UI provides for the same endpoint:

![image](https://github.com/scalar/scalar/assets/998703/5bf8b03b-7fdc-489c-93d4-f9f21fd2905b)

## Summary

Query parameters/variables names are missing :

- dropdown for known names
- eventually autocomplete would be nice

Query parameters/variables values are missing :

- dropdown for enums (eventually autocomplete would be nice)
- multi selection

## Conclusion

If this on the roadmap?

Can anyone point me in the right direction to write a PR?

**Comment by hanspagel** [comment]
Love it! We‘re in the process of revamping the whole API client, we’ll probably release it in the next days. This one is definitely on our list to add to the new API client.

It’s great that you're asking for a way to contribute. As we’re in the process of doing a new major release, it’s not the best moment to contribute to the API client, though. 😬 

I’ll keep you posted!

**Comment by JMLX42** [comment]
@hanspagel correct me if I am wrong but this is still an ongoing problem. Correct?

> It’s great that you're asking for a way to contribute. As we’re in the process of doing a new major release, it’s not the best moment to contribute to the API client, though. 😬

@hanspagel is that new release done? I can't find any release or tags.

**Comment by CHEDai** [comment]
@JMLX42 @hanspagel seems this is still not available yet?
